SCIP: populate new SymbolInformation fields

nicolas-guichard opened the pull request:
Fixes #15919

Veykril commented on crates/ide-db/src/defs.rs:
```diff
@@ -134,6 +143,132 @@ impl Definition {
+    pub fn label(
+        &self,
+        db: &RootDatabase,
+        famous_defs: Option<&FamousDefs<'_, '_>>,
```
This can be removed fully, its only used for its `db` field but the `db` is already being passed here

nicolas-guichard replied:
Indeed, I removed it.

Veykril commented:
One small thing, but other lgtm!

nicolas-guichard commented:
In addition to the change requested above, I've also rebased and added some tests for TraitAlias and TypeAlias which were not previously covered. I think the existing behavior for plain type aliases was unexpected (`type MyTypeAlias = u8;` yielded `[MyTypeAlias]` instead of `MyTypeAlias#`), but do tell me if you would rather revert to that.

Veykril commented:
Thanks!
@bors r+

bors commented:
:pushpin: Commit 73d9c77f2aeed394cf131dce55807be2d3f54064 has been approved by `Veykril`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust-analyzer) for this repository.

<!-- @bors r=Veykril 73d9c77f2aeed394cf131dce55807be2d3f54064 -->
<!-- homu: {"type":"Approved","sha":"73d9c77f2aeed394cf131dce55807be2d3f54064","approver":"Veykril","queue":"https://bors.rust-lang.org/homu/queue/rust-analyzer"} -->

bors commented:
:hourglass: Testing commit 73d9c77f2aeed394cf131dce55807be2d3f54064 with merge 078183231be8adb9bba07333e2819acc596a22cc...
<!-- homu: {"type":"BuildStarted","head_sha":"73d9c77f2aeed394cf131dce55807be2d3f54064","merge_sha":"078183231be8adb9bba07333e2819acc596a22cc"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang/rust-analyzer/actions/runs/7397543587/job/20125168159)
Approved by: Veykril
Pushing 078183231be8adb9bba07333e2819acc596a22cc to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"Veykril","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang/rust-analyzer/actions/runs/7397543587/job/20125168159"},"merge_sha":"078183231be8adb9bba07333e2819acc596a22cc"} -->
